### PR TITLE
adding new error response object

### PR DESCRIPTION
**Pull request by nitro-neal**

### CODE REVIEW

File: error/errorresponse.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+func NewErrorResponseWithError(err error, errorType ErrorType) *ErrorResponse {
+	return &ErrorResponse{
+		Valid:     isValid(errorType),
+		ErrorType: errorType,
+		Err:       err,
+	}
+}
+
+// GetErrorResponse will get the type of err, if the type is a ErrorResponse it will return that as an ErrorResponse, otherwise it will construct a new default ErrorResponse
+func GetErrorResponse(err error) ErrorResponse {
```
Comment (nitro-neal) [inline error/errorresponse.go:47]: This was not working for me:

var errorResponse ErrorResponse
errorBytes, e := json.Marshal(err)
if e != nil {
   return ErrorResponse{Valid: false, Err: err, ErrorType: UnknownError}
}
if e = json.Unmarshal(errorBytes, &errorResponse); err != nil {
   return ErrorResponse{Valid: false, Err: err, ErrorType: UnknownError}
}

File: credential/manifest/model_test.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func TestCredentialApplication(t *testing.T) {
 func TestCredentialResponse(t *testing.T) {
 	// example here https://identity.foundation/credential-manifest/#credential-response
 
-	t.Run("Credential Response - Fulfillment Vector 1", func(tt *testing.T) {
+	t.Run("Credential ErrorResponse - Fulfillment Vector 1", func(tt *testing.T) {
```
Comment (decentralgabe) [inline credential/manifest/model_test.go:101]: whoops

File: error/errorresponse.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	errResponseTypeString := reflect.TypeOf(ErrorResponse{}).String()
```
Comment (decentralgabe) [inline error/errorresponse.go:48]: ahh fun with reflection

File: credential/manifest/model_test.go [hunk 4]
```diff
@@ -100,7 +100,7 @@ func TestCredentialApplication(t *testing.T) {
...
```
Comment (decentralgabe) [inline credential/manifest/model_test.go:103]:
```suggestion
	t.Run("Credential Response - Fulfillment Vector 1", func(tt *testing.T) {
```

File: credential/manifest/model_test.go [hunk 5]
```diff
@@ -116,7 +116,7 @@ func TestCredentialResponse(t *testing.T) {
 		assert.JSONEq(tt, vector, string(roundTripBytes))
 	})
 
-	t.Run("Credential Response - Denial Vector 1", func(tt *testing.T) {
+	t.Run("Credential ErrorResponse - Denial Vector 1", func(tt *testing.T) {
```
Comment (decentralgabe) [inline credential/manifest/model_test.go:119]:
```suggestion
	t.Run("Credential Error - Denial Vector 1", func(tt *testing.T) {
```

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**